File: tenspiler/benchmark/cpp/color_dodge_8.cc
#include <iostream>
#include <vector>
#include <chrono>

#include "utils.h"

using namespace std;
using namespace std::chrono;

vector<vector<uint8_t>> color_dodge_8(vector<vector<uint8_t>> base, vector<vector<uint8_t>> active) {
    vector<vector<uint8_t>> out;
    int m = base.size();
    int n = base[0].size();
    for (int row = 0; row < m; row++) {
        vector<uint8_t> row_vec;
        for (int col = 0; col < n; col++) {
            uint8_t pixel;
            if (active[row][col] == 255)
                pixel = 255;
            else
                pixel = base[row][col] / (255 - active[row][col]);
            row_vec.push_back(pixel);
        }
        out.push_back(row_vec);
    }
    return out;
}


int main() {
    mat_timer(&color_dodge_8);
    return 0;
}

File: tenspiler/benchmark/python/color_dodge_8.py
from typing import List


def color_dodge_8(base: List[List[int]], active: List[List[int]]) -> List[List[int]]:
    output: List[List[int]] = []
    m = len(base)
    n = len(base[0])
    for i in range(m):
        curr_row: List[int] = []
        for j in range(n):
            if active[i][j] == 255:
                pixel = 255
            else:
                pixel = base[i][j] / (255 - active[i][j])
            curr_row.append(pixel)
        output.append(curr_row)
    return output
